pkg/aqhttp/resolv.go

```diff
@@ -0,0 +1,165 @@
+package aqhttp
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	TypeA    = 1  // IPv4
+	TypeAAAA = 28 // IPv6
+)
+
+type DoHResolver struct {
+	Server        string
+	Client        *http.Client
+	invalidRanges []*net.IPNet
+}
+
+func NewDoHResolver(server string) *DoHResolver {
+	if server == "" {
+		server = "https://1.1.1.1/dns-query"
+	}
+	ipv4Bogons := []string{
+		"0.0.0.0/8", "10.0.0.0/8", "100.64.0.0/10", "127.0.0.0/8",
+		"169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", "192.0.2.0/24",
+		"192.168.0.0/16", "198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24",
+		"224.0.0.0/4", "240.0.0.0/4", "255.255.255.255/32",
+	}
+
+	ipv6Bogons := []string{
+		"::/128",        // Unspecified
+		"::1/128",       // Loopback
+		"::ffff:0:0/96", // IPv4-mapped addresses
+		"100::/64",      // Discard prefix
+		"2001::/32",     // TEREDO
+		"2001:10::/28",  // Deprecated (ORCHID)
+		"2001:db8::/32", // Documentation
+		"fc00::/7",      // Unique local addresses (ULA)
+		"fe80::/10",     // Link-local
+		"ff00::/8",      // Multicast
+	}
+
+	ranges := append(ipv4Bogons, ipv6Bogons...)
+	var invalidRanges []*net.IPNet
+	for _, cidr := range ranges {
+		_, network, err := net.ParseCIDR(cidr)
+		if err == nil {
+			invalidRanges = append(invalidRanges, network)
+		}
+	}
```

Comment:
If this is just parsing static CIDRs, probably we should panic if this doesn't work? It'd imply there's an error in our static list of CIDRs